addons/http_routing/models/ir_http.py

```diff
@@ -703,5 +706,5 @@ def url_rewrite(self, path, query_args=None):
             new_url = path
         return new_url or path, endpoint and endpoint[0]
 
-    def _rewrite_len(self, website_id):
+    def _rewrite_len(self) -> int:
```

Comment:
pretty sure liskov-wise that it is invalid to drop the argument here

Reply:
I don't understand why? The argument is not used. It was a hack to pass a value from a getattr. `_rewrite_len` never needs to receive an argument.

But that's a good mistake because it means that without a website, the condition `request.env['ir.http']._rewrite_len() `is always false. I'm going to refactor and move this into website.

Reply:
> But that's a good mistake because it means that without a website, the condition request.env['ir.http']._rewrite_len() is always false. I'm going to refactor and move this into website.

Exactly what we expected! If you don't have website, you don't have website.rewrite, so you don't need to call heavy rewrite code... And if you have website but no website.rewrite (except redirect) it is the same...
But since it was not on ir.http before your commit, it was not overridable (as explain in comment) so it was an acceptable hack for perf gain, not a good mistake!

Reply:
@JKE-be Yes, my change was mistake/not good/incomplete. Before it was a hack (I didn't say it was a mistake before), but now it might as well be clean :)
So moving `_rewrite_len` into `website` seems better to me.